src/answers.sql

```diff
@@ -1,15 +1,118 @@
 -- Your answers here:
 -- 1
-
+SELECT type, SUM(mount) total_money FROM accounts GROUP BY type;
 -- 2
-
+SELECT COUNT(count) users_with_at_least_two_current_acounts FROM (
+    SELECT COUNT(user_id) count FROM accounts
+        WHERE type = 'CURRENT_ACCOUNT'
+        GROUP BY user_id
+        HAVING COUNT(user_id) > 1
+) AS match_users;
 -- 3
-
+SELECT id, type, user_id, mount FROM accounts ORDER BY mount DESC LIMIT 5;
 -- 4
-
+SELECT account_user_id, user_name, total_mount, (total_mount + SUM(final_mount)) final_mount
+    FROM (
+        SELECT u.id account_user_id, u.name user_name,
+            CASE
+                WHEN m.account_to = a.id THEN m.mount
+                WHEN m.type IN ('TRANSFER', 'OUT', 'OTHER') THEN m.mount * -1
+                ELSE m.mount
+            END final_mount
+        FROM users u
+            JOIN accounts a on u.id = a.user_id
+            JOIN movements m on a.id = m.account_from OR a.id = m.account_to
+    ) AS accounts_with_movements
+    JOIN (
+        SELECT user_id, SUM(mount) total_mount FROM accounts GROUP BY user_id
+    ) AS total_accounts_mounts ON total_accounts_mounts.user_id = account_user_id
+    GROUP BY account_user_id, user_name, total_mount
+    ORDER BY final_mount DESC
+    LIMIT 3;
 -- 5
+BEGIN TRANSACTION;
```

Comment:
good job working with function and transactions

Reply:
Thanks.